src/app/contexts/EventContext/index.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect } from 'react';
+import { node } from 'prop-types';
+import { useWindowEvent } from './useWindowEvent';
+import { useHandlerMap } from './useHandlerMap';
+
+export const EventContext = React.createContext({
+  useWindowEvent,
+  useClickTracker: () => {},
+});
+
+export const EventContextProvider = ({ children }) => {
+  const [handlerMap, setHandlerMap] = useState({});
+
+  const useClickTracker = (attr, handler) => {
```

Comment:
Whatas this doing? its a bit cryptic

Reply:
It takes a query selector, and handler function as arguments, so when a click is detected on the window, it makes sure the event's target matches the query selector, because invoking the handler function.
E.g.
```js
useClickTracker('[data-action-button="1"]', e => {
  // do something
});
```